tests/datasets/test_ase_datasets.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from ase import build, db
+from ase.io import write
+import os
+
+from ocpmodels.datasets import AseReadDataset, AseDBDataset
+
+structures = [
+    build.molecule("H2O", vacuum=4),
+    build.bulk("Cu"),
+    build.fcc111("Pt", size=[2, 2, 3], vacuum=8, periodic=True),
+]
+
+
+def test_ase_read_dataset():
+    for i, structure in enumerate(structures):
+        write(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), f"{i}.cif"
+            ),
+            structure,
+        )
+
+    dataset = AseReadDataset(
+        config={
+            "src": os.path.join(os.path.dirname(os.path.abspath(__file__))),
+            "pattern": "*.cif",
+        }
+    )
+
+    assert len(dataset) == len(structures)
+    data = dataset[0]
+
+    for i in range(len(structures)):
+        os.remove(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), f"{i}.cif"
+            )
+        )
+
+
+def test_ase_db_dataset():
+    try:
+        os.remove(
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), "asedb.db"
+            )
+        )
+    except FileNotFoundError:
+        pass
+
+    database = db.connect(
```

Comment:
I agree the logic looks good, but the test to show that rows add/delete/etc correctly works. It would be very easy to write a naive dataset that didn't implement this correctly. I added a test like that for the lmdb dataset.